openmdao/utils/general_utils.py

```diff
@@ -175,16 +174,21 @@ def ensure_compatible(name, value, shape=None, indices=None):
             shape = tuple(shape)
     elif not np.isscalar(value):
         shape = np.atleast_1d(value).shape
-    if indices is not None and indices.shaped_instance() is not None:
-        if indices.src_ndim > 1 and shape is None:
+
+    if indices is not None:  # and indices.shaped_instance() is not None:
+        if not indices._flat_src and shape is None:
             raise RuntimeError("src_indices for '%s' is not flat, so its input "
                                "shape must be provided." % name)
-        indshape = shape2tuple(indices.shape)
-        if shape is not None and indshape != shape:
-            raise ValueError("Shape of indices %s does not match shape of %s for '%s'." %
-                             (indshape, shape, name))
-        if shape is None:
-            shape = indshape
+        try:
+            indshape = indices.indexed_src_shape
+        except Exception:
```

Comment:
what kind of exception are we expecting here?  should be more specific...

Reply:
Could be different kinds (TypeError, ValueError, IndexError) so I just used a general exception.

Reply:
Perhaps it would be worth adding the tuple, so we could  pass this check:
https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html